app/javascript/utils/highlight.ts

```diff
@@ -49,7 +49,7 @@ function getLinesCount(text: string) {
 
 function wrapLineNumbers(code: string, start = 1) {
   const element = document.createElement('div')
-  element.innerHTML = code.trim()
+  element.innerHTML = code.replace(/(^[\r\n]+|[\r\n]+$)/g, '')
```

Comment:
This was causing the leading spaces/tabs to be removed. I've changed this to just removing newlines from the start and end.